### ISSUE TITLE
pysam.AlignmentFile not reading gzipped SAM files after v0.15.4

### ISSUE THREAD

**Issue by egaffo** [pysam-developers/pysam#1013]
pysam.AlignmentFile does not accept/read gzipped SAM files anymore since v0.15.4

**Comment by jmarshall** [comment]
It works for me in my testing. Please show an example program and data file that exhibit the problem you're seeing.

**Comment by egaffo** [comment]
```
$cat test.sam
@SQ     SN:10   LN:133797422
@SQ     SN:3    LN:198295559
@PG     ID:bwa  PN:bwa  VN:0.7.15-r1140 CL:bwa mem -t 8 -T 19 Homo_sapiens read_1.fastq.gz read_2.fastq.gz
A00153:590:H35TVDSXY:4:1101:28085:1000  83      10      81983381        60      140M    =       81983391        -130    GCTTTCTCTGATGGATGGGCTTCAAAATTGCTAATACACGAATTACTCATATTTAGACATTTGTTCTTTTAGGTGCCAGACTGTTTTCTTTTGGAATGTATTTTTGTGTGATGCAATTTGTCACCTAGATCTTAGCATTA  FFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF    NM:i:0  MD:Z:140        AS:i:140        XS:i:0
A00153:590:H35TVDSXY:4:1101:28085:1000  163     10      81983391        60      140M    =       81983381        130     ATGGATGGGCTTCAAAATTGCTAATACACGAATTACTCATATTTAGACATTTGTTCTTTTAGGTGCCAGACTGTTTTCTTTTGGAATGTATTTTTGTGTGATGCAATTTGTCACCTAGATCTTAGCATTAACTTTTAAAG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF    NM:i:0  MD:Z:140        AS:i:140        XS:i:20
A00153:590:H35TVDSXY:4:1101:9082:1000   83      3       116212781       60      140M    =       116212765       -156    TCATATGTACATAGGGTTACACTCTCACCCAACTACTTGAGAAGAATGCTACAAGGGGGATGGGAGCAGTAGACCACAGGGAATCTGATTTATTTGTCTCTGTAAGTACAAAGTTATACTATTTTACTGCAGTAAAGGTG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF    NM:i:0  MD:Z:140        AS:i:140        XS:i:19
A00153:590:H35TVDSXY:4:1101:9082:1000   163     3       116212765       60      65S75M  =       116212781       156     GGATCCCTGAAACCCTCTCTCTGTAGCATCATTCCTCCTCACTCATGGTTCCTGACACTCACAACTAACATAGGGTTACTTTCATATGTACATAGGGTTACACTCTCACCCAACTACTTGAGAAGAATGCTACAAGGGGG  FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF    NM:i:0  MD:Z:75 AS:i:75 XS:i:0  SA:Z:8,103429478,-,73S67M,60,0;

$gzip test.sam

$python

>>>import pysam
>>>pysam.AlignmentFile('test.sam.gz')

Segmentation fault (core dumped)
```

Tested with pysam v0.16.0.1 in a virtualenv with Python v3.6.9. Python2.7 is affected, too.
With pysam v0.15.4, there was no error.

**Comment by jmarshall** [comment]
Ah — I can't reproduce it with current development pysam, but I can with the 0.16.0.1 release and its associated htslib 1.10.2. This is an htslib bug that was fixed by samtools/htslib#1019, which appeared in htslib 1.11.

So it is already fixed in the pysam repository and the fix will appear in the next release. In the meantime, you could build pysam from Git or build the 0.16.0.1 release using `HTSLIB_MODE=external` etc.
